emerald-core/src/util/mod.rs

```diff
@@ -54,9 +53,8 @@ pub fn to_chain_name(id: u8) -> String {
 /// # Arguments:
 /// * `name` - target chain name
 ///
-pub fn to_chain_id(name: &str) -> u8 {
+pub fn to_chain_id(name: &str) -> usize {
```

Comment:
`usize` is platform specific and can be a source of problems on embeddable environments (16-, 8-bit).